### PR TITLE
setup new testrunner components

### PR DESCRIPTION
**Pull request by tokle**
This will setup required components when executing tests.
Next step: invoke the correct test runner based on compatibility

### CODE REVIEW

File: vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/JunitHandler.java [hunk 1]
```diff
@@ -2,70 +2,77 @@
 package com.yahoo.vespa.testrunner;
 
 import ai.vespa.hosted.api.TestDescriptor;
-import ai.vespa.hosted.cd.internal.TestRuntimeProvider;
+import com.google.common.base.Strings;
```
Comment (bjorncs) [inline vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/JunitHandler.java:5]: Don't use Guava unless really necessary

File: vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/JunitRunner.java [hunk 2]
```diff
@@ -55,27 +60,43 @@ public JunitRunner(OsgiFramework osgiFramework) {
...
+
+    public boolean isSupported() {
+        Bundle testBundle = findTestBundle();
+        return loadTestDescriptor(testBundle).isPresent();
+    }
+
+    private Bundle findTestBundle() {
         return Stream.of(bundleContext.getBundles())
-                .filter(bundle -> bundle.getSymbolicName().endsWith(bundleNameSuffix))
+                .filter(bundle -> bundle.getSymbolicName().endsWith(TEST_BUNDLE_SUFFIX))
```
Comment (bjorncs) [inline vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/JunitRunner.java:81]: Bundle symbolic name no longer contains `-tests` suffix. I will add it back if necessary.

Comment (bjorncs) [reply vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/JunitRunner.java:81]: Iterate through all bundles and look for `X-JDisc-Test-Bundle-Version` header

File: cloud-tenant-cd/src/main/java/ai/vespa/hosted/cd/cloud/impl/package-info.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+/**
+ * @author mortent
+ */
+@ExportPackage
```
Comment (bjorncs) [inline cloud-tenant-cd/src/main/java/ai/vespa/hosted/cd/cloud/impl/package-info.java:6]: Is this needed?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 4]
```diff
@@ -855,12 +855,26 @@ static NodeResources testerResourcesFor(ZoneId zone, DeploymentInstanceSpec spec
...
+                ? "vz-tenant-cd"
+                : "cloud-tenant-cd";
 
         String servicesXml =
                 "<?xml version='1.0' encoding='UTF-8'?>\n" +
                 "<services xmlns:deploy='vespa' version='1.0'>\n" +
                 "    <container version='1.0' id='tester'>\n" +
                 "\n" +
+                // Test runners using container and osgi
+                "        <component id=\"" + testRuntime + "\" bundle=\"" + testRuntimeBundle + "\" />\n" +
```
Comment (bjorncs) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java:871]: @gjoranv  Do you know about a better approach?

File: config-model/src/main/java/com/yahoo/vespa/model/container/ApplicationContainerCluster.java [hunk 5]
```diff
@@ -140,6 +140,8 @@ private void addTestrunnerComponentsIfTester(DeployState deployState) {
             addPlatformBundle(Paths.get(Defaults.getDefaults().underVespaHome("lib/jars/tenant-cd-api-jar-with-dependencies.jar")));
             if(deployState.zone().system().isPublic()) {
                 addPlatformBundle(Paths.get(Defaults.getDefaults().underVespaHome("lib/jars/cloud-tenant-cd-jar-with-dependencies.jar")));
+            } else {
+                addPlatformBundle(Paths.get(Defaults.getDefaults().underVespaHome("lib/jars/vz-tenant-cd-jar-with-dependencies.jar")));
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ApplicationContainerCluster.java:144]: This is already done in the amender

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 6]
```diff
@@ -855,12 +855,26 @@ static NodeResources testerResourcesFor(ZoneId zone, DeploymentInstanceSpec spec
         String resourceString = String.format(Locale.ENGLISH,
                                               "<resources vcpu=\"%.2f\" memory=\"%.2fGb\" disk=\"%.2fGb\" disk-speed=\"%s\" storage-type=\"%s\"/>",
                                               resources.vcpu(), resources.memoryGb(), resources.diskGb(), resources.diskSpeed().name(), resources.storageType().name());
+        String testRuntime = systemUsesAthenz
+                ? "ai.vespa.hosted.cd.vz.impl.VespaTestRuntimeProvider"
+                : "ai.vespa.hosted.cd.cloud.impl.VespaTestRuntimeProvider";
+        String testRuntimeBundle = systemUsesAthenz
+                ? "vz-tenant-cd"
```
Comment (bjorncs) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java:862]: Consider moving these constants to controller config

**Review by bjorncs — APPROVED** [review]

File: cloud-tenant-cd/src/main/java/ai/vespa/hosted/cd/cloud/impl/package-info.java (hunk 3 above)
Comment (gjoranv) [reply cloud-tenant-cd/src/main/java/ai/vespa/hosted/cd/cloud/impl/package-info.java:6]: In general, exporting a package is only necessary if another bundle should use your apis programmatically. It's not needed for a component to be constructed.

**Comment by tokle** [comment]
Closed in favor of #13788

**Pull request closed without merging**